packages/grafana-ui/src/components/Input/Input.tsx

```diff
@@ -225,6 +237,21 @@ export const Input = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
 
   const theme = useTheme();
   const styles = getInputStyles({ theme, invalid: !!invalid, width });
+  const innerRef = useRef<HTMLInputElement>(null);
+
+  const onNumberInputBlur = useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      if (isNaN(parseInt(e.target.value, 10))) {
+        // Force UI change for Firefox and Safari issue #29207
+        ((ref as RefObject<HTMLInputElement>) || innerRef).current!.value = '';
```

Comment:
Do we need the ref here if we already have access to the element value via `event.target.value`?

Reply:
Indeed. we don't need ref, thanks.